ipc/nodepart/arwenDriver.go

```diff
@@ -0,0 +1,308 @@
+package nodepart
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path"
+	"strings"
+	"syscall"
+
+	"github.com/ElrondNetwork/arwen-wasm-vm/ipc/common"
+	"github.com/ElrondNetwork/arwen-wasm-vm/ipc/logger"
+	vmcommon "github.com/ElrondNetwork/elrond-vm-common"
+)
+
+var _ vmcommon.VMExecutionHandler = (*ArwenDriver)(nil)
+
+// ArwenDriver manages the execution of the Arwen process
+type ArwenDriver struct {
+	nodeLogger     logger.Logger
+	blockchainHook vmcommon.BlockchainHook
+	vmType         []byte
+	blockGasLimit  uint64
+	gasSchedule    map[string]map[string]uint64
+
+	arwenInputRead   *os.File
+	arwenInputWrite  *os.File
+	arwenOutputRead  *os.File
+	arwenOutputWrite *os.File
+	arwenLogRead     *os.File
+	arwenLogWrite    *os.File
+	command          *exec.Cmd
+	part             *NodePart
+}
+
+// NewArwenDriver creates a new driver
+func NewArwenDriver(
+	nodeLogger logger.Logger,
+	blockchainHook vmcommon.BlockchainHook,
+	vmType []byte,
+	blockGasLimit uint64,
+	gasSchedule map[string]map[string]uint64,
+) (*ArwenDriver, error) {
+	driver := &ArwenDriver{
+		nodeLogger:     nodeLogger,
+		blockchainHook: blockchainHook,
+		vmType:         vmType,
+		blockGasLimit:  blockGasLimit,
+		gasSchedule:    gasSchedule,
+	}
+
+	err := driver.startArwen()
+	return driver, err
+}
+
+func (driver *ArwenDriver) startArwen() error {
+	driver.nodeLogger.Info("ArwenDriver.startArwen()")
+
+	driver.resetPipeStreams()
+
+	arwenPath, err := driver.getArwenPath()
+	if err != nil {
+		return err
+	}
+
+	arguments, err := common.PrepareArguments(driver.vmType, driver.blockGasLimit, driver.gasSchedule, logger.LogDebug)
```

Comment:
this preparation should be done only at constructor time. and the results to be saved in arwenDriver.

Reply:
@sasurobert I will fix this according to the chose option at:

https://github.com/ElrondNetwork/arwen-wasm-vm/pull/78#discussion_r392863036

Because moving parameter preparation logic means also moving temp file creation logic. And that affects temp file deletion logic (`ParseArguments()` in `arguments.go`).

If we choose to pass the gas schedule `base64(...)`, no more file creation & deletion needed, so we can move this preparation at constructor time easily.

So?